EpsilonAuxBmsV2/Core/Src/main.c

```diff
@@ -0,0 +1,893 @@
+    GPIO_InitStruct.Pull = GPIO_PULLUP;
+    HAL_GPIO_Init(GPIOE, &GPIO_InitStruct);
+
+}
+
+/* USER CODE BEGIN 4 */
+void MX_CAN1_User_Init(void)
+{
+    CAN_FilterTypeDef orionVoltageTempFilterConfig;
+    orionVoltageTempFilterConfig.FilterBank = 0; // Use first filter bank
+    orionVoltageTempFilterConfig.FilterMode = CAN_FILTERMODE_IDLIST; // Look for specific can messages
+    orionVoltageTempFilterConfig.FilterScale = CAN_FILTERSCALE_32BIT;
+    orionVoltageTempFilterConfig.FilterIdHigh = ORION_MAX_MIN_VOLTAGES_STDID << 5; // Filter registers need to be shifted left 5 bits
+    orionVoltageTempFilterConfig.FilterIdLow = 0;
+    orionVoltageTempFilterConfig.FilterMaskIdHigh = ORION_TEMP_INFO_STDID << 5;
+    orionVoltageTempFilterConfig.FilterMaskIdLow = 0; // Unused
+    orionVoltageTempFilterConfig.FilterFIFOAssignment = 0;
+    orionVoltageTempFilterConfig.FilterActivation = ENABLE;
+    orionVoltageTempFilterConfig.SlaveStartFilterBank = 0; // Set all filter banks for CAN1
+
+    if (HAL_CAN_ConfigFilter(&hcan1, &orionVoltageTempFilterConfig) != HAL_OK)
+    {
+        /* Filter configuration Error */
+        Error_Handler();
+    }
+
+    CAN_FilterTypeDef orionPackInfoFilterConfig;
+    orionPackInfoFilterConfig.FilterBank = 1; // Use secondary filter bank
+    orionPackInfoFilterConfig.FilterScale = CAN_FILTERSCALE_32BIT;
+    orionPackInfoFilterConfig.FilterMode = CAN_FILTERMODE_IDLIST; // Look for specific can messages
+    orionPackInfoFilterConfig.FilterIdHigh = ORION_PACK_INFO_STDID << 5; // Filter registers need to be shifted left 5 bits
+    orionPackInfoFilterConfig.FilterIdLow = 0; // Filter registers need to be shifted left 5 bits
+    orionPackInfoFilterConfig.FilterMaskIdHigh = 0;
+    orionPackInfoFilterConfig.FilterMaskIdLow = 0; // Unused
+    orionPackInfoFilterConfig.FilterFIFOAssignment = 0;
+    orionPackInfoFilterConfig.FilterActivation = ENABLE;
+    orionPackInfoFilterConfig.SlaveStartFilterBank = 0; // Set all filter banks for CAN2
+
+    if (HAL_CAN_ConfigFilter(&hcan1, &orionPackInfoFilterConfig) != HAL_OK)
+    {
+        /* Filter configuration Error */
+        Error_Handler();
+    }
+}
+
+void HAL_CAN_RxFifo0MsgPendingCallback(CAN_HandleTypeDef* hcan)
+{
+    CAN_RxHeaderTypeDef hdr;
+    uint8_t             data[8] = {0};
+
+    if (HAL_CAN_GetRxMessage(hcan, CAN_RX_FIFO0, &hdr, data) != HAL_OK)
+    {
+        return;
+    }
+
+    CanRxQueueData message;
+    message.canRxHeader = hdr;
+    memcpy(message.data, data, sizeof(uint8_t) * 8);
+    osMessageQueuePut(canRxParserQueue, &message, 0, IRQ_QUEUE_PUT_TIMEOUT);
+}
+
+void HAL_SPI_RxCpltCallback(SPI_HandleTypeDef* hspi)
+{
+    osThreadFlagsSet(readAuxVoltageTaskHandle, SPI_READY_FLAG);
+}
+
+void vApplicationMallocFailedHook( void )
+{
+    errorCode = MALLOC_FAILED;
+    Error_Handler();
+}
+
+void vApplicationStackOverflowHook( TaskHandle_t* pxTask, signed char* pcTaskName )
+{
+
+    if (*pxTask == startupTaskHandle)
```

Comment:
We kept getting this error: "switch quantity not an integer." And since the underlying type of `TaskHandle_t` is a `void*`, we had to go with this mess